src/sagemaker/jumpstart/types.py

```diff
@@ -1137,21 +1142,43 @@ def from_json(self, json_obj: Dict[str, Any]) -> None:
         if json_obj is None:
             return
 
-        self.preset_configs: str = {
+        self.preset_configs: Dict[str, JumpStartPresetConfig] = {
             alias: JumpStartPresetConfig(preset_config)
             for alias, preset_config in json_obj["preset_configs"].items()
         }
-        self.preset_config_rankings: str = json_obj["preset_config_rankings"]
+        self.preset_config_rankings: Optional[Dict[str, JumpStartPresetRanking]] = json_obj[
+            "preset_config_rankings"
+        ]
 
     def to_json(self) -> Dict[str, Any]:
         """Returns json representation of JumpStartInferencePresetConfig object."""
         json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
         return json_obj
 
-    def get_resolved_config(self) -> JumpStartPresetConfig:
-        """Resolves the preset config."""
-        # TODO: return the best preset config based on config rankings
-        return self.preset_configs[0] if self.preset_configs else None
+    def get_top_config_from_ranking(
+        self,
+        ranking_name: str = "overall",
```

Comment:
maybe define `overall` as a constant somewhere?

Reply:
There's a circular dependency to put it in `constants.py`, so I created a enum to store it